### PR TITLE
#DP-75 fix: Pagination

### PR DESCRIPTION
**Pull request by jeandamascyifuzo**
### PR Description
This pull request add the pagination to the application

### Description of tasks that were expected to be completed

- [x] Pagination should be working 
- [x] All pages that needs paginated data are worked on

### Background context
Since the task is around managing data already structure in tables, the `react-table` library was introduced to ensure that we maintain the uniformity of the application.
This has required an over-hall changes to the implementation of the data rendering while adding more functionalities (offered by react-table out of the box) like
- Column sorting
- Global filtering
- Advanced pagination

### How has this been tested?
```
git clone https://github.com/atlp-rwanda/atlp-pulse-fn.git
cd atlp-pulse-fn
git checkout ft-pagination
npm i
npm run dev
```
### Or
Use this[ link ](https://devpulse-review-app-75.herokuapp.com/dashboard/manage)to test

### CODE REVIEW

**Comment by feyton** [comment]
I deployed a review app for you to try out this pull request 👇🏽

Heroku App: https://devpulse-review-app-75.herokuapp.com/


<hidden purpose="for-rewritable-pr-comment-action-use" value="Deploy"></hidden>

**Review by rukundo-kevin — CHANGES_REQUESTED** [review]
The trainee  and ratings tables still don't have pagination 
Screen Shot 2022-08-31 at 09 17 56

**Review by chanceDira — APPROVED** [review]
Good work @jeandamascyifuzo. You can increase your test coverage to make this ready.

**Review by Procech20 — APPROVED** [review]

**Review by rukundo-kevin — CHANGES_REQUESTED** [review]
Great work @jeandamascyifuzo , but still a few tables to add pagination to.
Some tables when you are logged in as Trainee  don't have paginations
 - Performance table
 - Sessions table
And when you are logged in as a coordinator the Ratings table don't have pagination

Screen Shot 2022-09-01 at 11 22 30

File: src/containers/Routes.tsx [hunk 1]
```diff
@@ -26,6 +27,7 @@ function MainRoutes() {
           <Route path="/pricing" element={<Pricing />} />
           <Route path="/noredirect" element={<Noredirect />} />
           <Route path="/pricing-form" element={<Pay />} />
+          {/* <Route path="/test" element={<Test />} /> */}
```
Comment (feyton) [inline src/containers/Routes.tsx:30]: Remove this line

File: src/containers/admin-dashBoard/ManagerRoles.tsx [hunk 2]
```diff
@@ -269,69 +278,130 @@ const AdminSission = () => {
...
-                                </span>
-                                <div className="flex flex-col  leading-4 px-3 py-2">
-                                  <h3 className="dark:text-white sm:text-xs">
-                                    {developer.name}
-                                  </h3>
-                                  <p className="text-sm sm:text-xs text-gray-400 dark:text-white">
-                                    {developer.email}
-                                  </p>
+                          let Data;
+                          if (
```
Comment (feyton) [inline src/containers/admin-dashBoard/ManagerRoles.tsx:290]: Please re-check the way you rebased. You have affected previous features of @chanceDira

File: src/pages/AdminTraineeDashboard.tsx [hunk 3]
```diff
@@ -209,52 +224,116 @@ const AdminTraineeDashboard = () => {
...
-                                let rowTheme =
-                                  index % 2 !== 0
-                                    ? 'bg-light-bg dark:bg-dark-tertiary'
-                                    : 'bg-white dark:bg-dark-bg';
+                              {developers
+                                .slice(firstContentIndex, lastContentIndex)
+                                .map((developer: any, index: any) => {
+                                  let rowTheme =
+                                    index % 2 !== 0
+                                      ? 'bg-light-bg dark:bg-dark-tertiary'
```
Comment (feyton) [inline src/pages/AdminTraineeDashboard.tsx:232]: Same rebase issue happened here

Comment (feyton) [reply src/pages/AdminTraineeDashboard.tsx:232]: From line 232 to 247 should no be there

**Review by feyton — CHANGES_REQUESTED** [review]
@jeandamascyifuzo 

Please check the comments I left

**Review by rukundo-kevin — APPROVED** [review]

**Review by ceelogre — CHANGES_REQUESTED** [review]
Filtering is affected by pagination. Take a look at the video.
https://user-images.githubusercontent.com/5022823/188084347-4aebea19-52c8-4c27-9dbc-49b104e2dab3.mov

**Comment by jeandamascyifuzo** [comment]
> 

@ceelogre 
Right now, I am not aware of how I can implement it, might take some time

**Review by feyton — APPROVED** [review]

**Review by Dev-nkundineza — CHANGES_REQUESTED** [review]
Screen Shot 2022-09-07 at 12 56 57

Great work @jeandamascyifuzo , can we improve UX in the light theme.

**Review by chanceDira — APPROVED** [review]

**Review by Dev-nkundineza — DISMISSED** [review]
Screen Shot 2022-09-07 at 14 33 25

The cohort list has no pagination, 

Screen Shot 2022-09-07 at 12 56 57
The above is on the ratings, when logged in as coordinator

**Review by Dev-nkundineza — APPROVED** [review]
Good Work

**Pull request merged**